packages/bson-transpilers/symbols/php/templates.yaml

```diff
@@ -0,0 +1,1231 @@
+                if (
+                    (str.charAt(0) === '\'' && str.charAt(str.length - 1) === '\'') ||
+                    (str.charAt(0) === '"' && str.charAt(str.length - 1) === '"')
+                ) {
+                    str = str.substr(1, str.length - 2);
+                }
+                return `${str.replace(/\\([\s\S])|(')/g, '\\$1$2')}`;
+            };
+        
+            return `'/${stringifyWithSingleQuotes(pattern)}/${flags ? flags : ''}'`;
+        }
+    BoolTypeTemplate: &BoolTypeTemplate !!js/function >
+        (literal) => {
+            return literal.toLowerCase();
+        }
+    IntegerTypeTemplate: &IntegerTypeTemplate null # args: literal, argType (i.e. '_string', '_decimal' etc)
+    DecimalTypeTemplate: &DecimalTypeTemplate null # args: literal, argType
+    LongBasicTypeTemplate: &LongBasicTypeTemplate null # args: literal, argType
+    HexTypeTemplate: &HexTypeTemplate null # args: literal, argType
+    OctalTypeTemplate: &OctalTypeTemplate !!js/function >
+        (literal) => {
+            let offset = 0;
+
+            if (
+                literal.charAt(0) === '0' &&
+                (literal.charAt(1) === '0' || literal.charAt(1) === 'o' || literal.charAt(1) === 'O')
+            ) {
+                offset = 2;
+            } else if (literal.charAt(0) === '0') {
+                offset = 1;
+            }
+
+            literal = `0${literal.substr(offset, literal.length - 1)}`;
+
+            return literal;
+        }
+    NumericTypeTemplate: &NumericTypeTemplate null # args: literal, argType
+    ArrayTypeTemplate: &ArrayTypeTemplate !!js/function >
+        (literal, depth) => {
+            if (literal === '') {
+                return '[]'
+            }
+            return `[${literal}]`;
+        }
+    ArrayTypeArgsTemplate: &ArrayTypeArgsTemplate null # Args: single array element, nestedness, lastElement? (note: not being used atm)
+    NullTypeTemplate: &NullTypeTemplate !!js/function >
+        () => {
+            return 'null';
+        }
+    UndefinedTypeTemplate: &UndefinedTypeTemplate !!js/function >
+        () => {
+            return 'null';
+        }
+    ObjectTypeTemplate: &ObjectTypeTemplate !!js/function >
+        (literal) => {
+            return `${literal}`;
+        }
+    ObjectTypeArgsTemplate: &ObjectTypeArgsTemplate !!js/function >
+        (args, depth) => {
+            let isObj = true;
+        
+            if (args.length === 0) {
+                return `[]`;
+            }
+
+            const checkIndex = (strIndex, numericIndex) => {
+                let parsedIndex = parseInt(strIndex);
+                if (isNaN(parsedIndex) || ('' + parsedIndex !== strIndex) || parsedIndex != numericIndex) return false;
+                return true;
+            }
+        
+            let indexTest = -1;
+            let pairs = args.map((arg) => {
+                indexTest++;
+                if (isObj && !checkIndex(arg[0], indexTest)) {
```

Comment:
If you think it's more readable, feel free to initialize `indexTest` to zero and then either use `indexTest++` here or perform the increment _after_ this conditional and before the `return`. Otherwise, this can be left as-is.

Reply:
Fixed.